content/blog/2022/10/volumemounts-policy.md

```diff
@@ -0,0 +1,104 @@
+
+You can find it [published in ArtifactHub.io](https://artifacthub.io/packages/kubewarden/volumemounts-policy/volumemounts).
+As usual, its artifact is signed with Sigstore in keyless mode, and if you are
+curious, you can peek into the policy's
+[implementation in Rust here](https://github.com/kubewarden/volumemounts-policy).
+
+This new policy joins the already existing
+[volumes-psp policy](https://artifacthub.io/packages/kubewarden/volumes-psp/volumes-psp),
+which provides an allowlist of volume types, and
+[hostpaths-psp policy](https://artifacthub.io/packages/kubewarden/hostpaths-psp/hostpaths-psp),
+with an allowlist of hostPath volumes.
+
+
+### What is so useful about the new volumeMounts policy?
+
+The existing PSP policies restricted usage of volumes, as a cluster admin. The
+new volumeMounts policy has settings with 4 operators that enable you to check
+for compliance and migration usecases, even if you don't have access to
+controlling creation of Volumes in the cluster. Let's see them: 
+
+- `reject: anyIn`: Works as a denylist of your usage of volumes.
+  Since we are checking volume names, we can also filter
+  [Out-Of-Tree volumes](https://kubernetes.io/docs/concepts/storage/volumes/#out-of-tree-volume-plugins):
+    ```yaml
+    reject: anyIn
+    volumeMountsNames:
+    - my-secure-hostpath-volume
+    - my-cache-volume
+    - my-out-of-tree-volume
+    ```
+- `reject: anyNotIn`: Works as an allowlist.
+    ```yaml
+    reject: anyNotIn
+    volumeMountsNames:
+    - my-secrets-volume
+    - my-volume2
+    ```
+- `reject: allAreUsed`: The container cannot use all listed volumes at once.
+  Helpful for enforcing migration between volumes, for example:
+    ```yaml
+    reject: allAreUsed
+    volumeMountsNames:
+    - old-deprecated-volume
+    - new-supported-volume
+    ```
+
+- `reject: notAllAreUsed`: The container can use all listed volumes at once, but only one of them.
+  Helpful for enforcing backup operations, for example:
+    ```yaml
+    reject: notAllAreUsed
+    volumeMountsNames:
+    - work-volume
+    - backup-volume
+    ```
+
+
+### In action
+
+Just instantiate an `AdmissionPolicy` or cluster-wide `ClusterAdmissionPolicy`
+with the policy module and settings. Here's a definition of a policy that
+rejects any workload resource (Pods, Deployments, Cronjobs..) that doesn't
+adhere to the `volumeMounts` allowlist:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: policies.kubewarden.io/v1
+kind: ClusterAdmissionPolicy
+metadata:
+  name: volumemounts-policy
+spec:
+  module: ghcr.io/kubewarden/policies/volumemounts-policy:v0.1.0
+  mutating: false
+  rules:
+  - apiGroups: [""]
+    apiVersions: ["v1"]
+    resources: ["pods", "deployments", "replicasets", "daemonsets", "replicationcontrollers", "jobs", "cronjobs"]
+    operations:
+    - CREATE
+    - UPDATE
+  settings:
+    reject: anyNotIn # as an allowlist
+    volumeMounts:
+    - my-volume
+    - my-volume2
+EOF
+```
+
```

Comment:
I think it would be useful to provide an example of a `Pod` that would be rejected, and add the error message it would display.